lib/context.js

```diff
@@ -52,42 +60,65 @@ module.exports = async function getContext() {
 
       const actor = response.data.display_name || 'Unknown User';
 
-      return {
-        ref: process.env.BITBUCKET_TAG
-          ? `refs/tags/${process.env.BITBUCKET_TAG}`
-          : `refs/heads/${process.env.BITBUCKET_BRANCH}`,
-        sha: process.env.BITBUCKET_COMMIT,
-        actor,
-        runId: parseInt(process.env.BITBUCKET_BUILD_NUMBER, 10),
-        payload: {
-          repository: {
-            name: process.env.BITBUCKET_REPO_SLUG,
-            organization: process.env.BITBUCKET_WORKSPACE,
-          },
-          commits: [
-            {
-              message: execSync('git log --format=%B -n 1 $BITBUCKET_COMMIT', {
-                env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
-                encoding: 'utf-8',
-              }).trim(),
-              // Get the unix timestamp of the commit then convert it to
-              // an ISO8601 UTC string
-              //
-              // https://git-scm.com/docs/pretty-formats#Documentation/pretty-formats.txt-emctem
-              // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toISOString
-              timestamp: new Date(
-                execSync('git show -s --format=%ct $BITBUCKET_COMMIT', {
-                  env: { BITBUCKET_COMMIT: process.env.BITBUCKET_COMMIT },
-                  encoding: 'utf-8',
-                }).trim() * 1000
-              ).toISOString(),
-              author: {
-                name: actor,
+      // BITBUCKET_PR_ID only set on pull request triggered builds
+      // this is how we determine if it's a `push` or `pull_request`
+      const eventName = process.env.BITBUCKET_PR_ID ? 'pull_request' : 'push';
+
+      switch (eventName) {
+        case 'push':
+          return {
+            ...baseContext,
+            eventName,
+            payload: {
+              repository: {
+                name: process.env.BITBUCKET_REPO_SLUG,
+                // TODO we should remove this in a future release
+                organization: process.env.BITBUCKET_WORKSPACE,
+              },
+              organization: {
+                login: process.env.BITBUCKET_WORKSPACE,
               },
+              commits: [
```

Comment:
[docs for GH Webhooks](https://docs.github.com/en/webhooks-and-events/webhooks/webhook-events-and-payloads#push) says this is an array, not a singular commit.

> An array of commit objects describing the pushed commits. (Pushed commits are all commits that are included in the `compare` between the `before` commit and the `after` commit.)

see comment below

Reply:
Yup, aware of this. But in micro we're currently only saving the top commit anyway! So this is fine for now.

Reply:
See: https://github.com/readmeio/readme-micro/pull/23#discussion_r1303973578